src/grafana_agent.py

```diff
@@ -868,6 +868,88 @@ def _receiver_config(protocol: str):
 
         return config
 
+    @property
+    def _tracing_sampling(self) -> Dict[str, Any]:
+        return {
+            "policies": [
+                {
+                    "name": "error-traces-policy",
+                    "type": "and",
+                    "and": {
+                        "and_sub_policy": [
+                            {
+                                "name": "trace-status-policy",
+                                "type": "status_code",
+                                "status_code": {"status_codes": ["ERROR"]},
```

Comment:
I don't understand how this works, will this also correctly filter charm traces where an exception has been raised? what is status_code? does it apply to http requests, or is that an attribute that gets set on all otlp traces and error traces will have it by default?

Reply:
You're right, this is tricky. I'll write more comments in the code around this.

Tail sampling processor for `status_code` uses span status: https://opentelemetry.io/docs/concepts/signals/traces/#span-status - which seems to be interpreted within the sampling processor as "if within the trace there is a span that has an `error` status, return that trace if you're looking for error traces". That is also how Tempo search interprets it based on my tests.

Reply:
Added comments that hopefully explain it.